### DISCUSSION TITLE
How to go back, if scaffold deletes Topo Faces and Topo Edges

### DISCUSSION THREAD

**Discussion by dkowollik** [ansys/pyprimemesh#863] [Q&A]
Hi, 
I have a large slender model with large differences in dimensionality. Therefore, I have to play around with tolerances on the Scaffold operations. Up to now, I always have to reopen the large model, which always takes approx. 30s. Is it possible to go back one step? Or do you have an alternative solution to keep multiple models or parts?

**Comment by waltersma** [comment]
Hi @dkowollik,

What CAD format are you importing? 

If you are not already then it would likely be quicker if you exported as FMD format from Space claim/Discovery and read FMD natively. 

If you are using other formats this may be slower as non native formats will launch out of process reader applications to import.

Or you can import once and write as PMDAT. Subsequent reads from PMDAT after the import has already been done.

I hope this helps.

Thanks

**Reply by dkowollik** [reply]
Hi @waltersma,

thank you for your answer. My native format is Rhino *.3dm. I also tried other formats like Step or imported *.3dm with the GeometryService and exported it as FMD. These formats are not faster than importing the *.3dm file. 
Would it be possible to copy the parts somehow in memory and attach it to the model, so that I can work with one part on a certain part id, play around with scaffold options and reuse the original part by copying it again? If this is an option, can you provide python snippets to do this?

**Reply by waltersma** [reply]
Hi @dkowollik, 

Copying in this way topo part objects is currently not exposed by the API.  Have you considered saving a subset of the model as a file while you are experimenting rather than reading the entire model each time?

Thanks

**Reply by dkowollik** [reply]
Hi @waltersma, 

At the moment I am not even able to create the labels I need. Therefore, I am not able to select the desired topo faces. I tried the fmd format for the entire model, but this is not any faster than step or *.3dm files. Could you please provide an example script to save only portions of a part? As soon as I can select the desired topo faces, I can give it a try.
